qtmodern/windows.py

```diff
@@ -169,6 +168,39 @@ def setWindowTitle(self, title):
         super(ModernWindow, self).setWindowTitle(title)
         self.lblTitle.setText(title)
 
+    def _setWindowButtonState(self, button: QToolButton, state: bool):
+        button.setVisible(state)
```

Comment:
It would be great if we can add logic to disable the buttons instead of hidding them if not _all_ of the flags are set to false. It can be a bit confusing for the user if only a few buttons are hidden.

Reply:
And we would then also need to add to the css to indicate that they are disabled
```
#btnClose::disabled, #btnRestore::disabled, #btnMaximize::disabled, #btnMinimize::disabled {
  background-color: palette(midlight);
}
```

Reply:
This is not how the implementation in standard qt5 is tho. If I do `mw.setWindowFlag(Qt.WindowCloseButtonHint, False)`, it will only hide the close button (tested only on linux with KDE Plasma).

Reply:
That is interesting, this is how it behaves on both windows and macOS with only one button disabled.
![Screenshot 2020-02-10 at 14 20 09](https://user-images.githubusercontent.com/5273682/74153303-a040a400-4c10-11ea-9ccb-2e2cbe7c0a97.png)

This is how it looks with all of them disabled
![Screenshot 2020-02-10 at 14 23 58](https://user-images.githubusercontent.com/5273682/74153616-51473e80-4c11-11ea-9585-8bc5a8d4975d.png)

Reply:
On KDE Plasma:

https://i.imgur.com/cR2HPnf.png

https://i.imgur.com/G5xmyfm.png

Reply:
Most people are on macOS or Windows, so one of them is probably better. Of them I think either would be fine. The macOS version is probably the easiest to implement so let us go with that one, or what do you think?

Reply:
There should also be some logic to handle the case of `maximize` and `restore`. Their visibility should also different depending if the window is maximized or not.